extension/ts/services/finder/impl/InternetArchive.ts

```diff
@@ -1,7 +1,41 @@
 import {Finder, FinderResponse} from "../finder";
+import {invokeWorker} from "../../../workers/invoker";
+import {WorkerKind} from "../../../workers/types";
 
-const InternetArchive: Finder = async (): Promise<FinderResponse> => {
-	return [];
+const MAX = 3;
+
+const BASE_URL = "https://archive.org";
+const SEARCH_PATH = "/advancedsearch.php";
+const DETAILS_PATH = "/details";
+
+const toLink = (response: any): string => {
+	if (!!response && typeof response === "object" && "identifier" in response) {
+		const {identifier} = response;
+		return `${BASE_URL}${DETAILS_PATH}/${identifier}`;
+	} else {
+		return null;
+	}
+
+};
+
+const getLinks = (responseString: string): string[] => {
+	const response = JSON.parse(responseString).response;
+	return response.docs.map(toLink).filter((link) => !!link);
+};
```

Comment:
we're making a new `getLinks` specifically for InternetArchive because the one in `getLinks.ts` won't work here?

Reply:
oh i should have replied

the other one scrapes a webpage. this one uses a proper API with a JSON response!